app/src/main/java/com/example/instasnap/Register.java

```diff
@@ -0,0 +1,164 @@
+        // Anonymous Variable Example: One of the two listener type implementations requested
+        _loginHereTextView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent loginIntent = new Intent(getApplicationContext(), Login.class);
+                startActivity(loginIntent);
+                finish();
+            }
+        });
+    }
+
+
+    // Member Class Example: One of the two listener type implementations requested
+    private class AuthCompleteListener implements OnCompleteListener<AuthResult> {
+        private final Register _registerActivity;
+
+        public AuthCompleteListener(Register activity) {
+            this._registerActivity = activity;
+        }
+
+        @Override
+        public void onComplete(@NonNull Task<AuthResult> task) {
+            _registerActivity.onAuthenticationComplete(task);
+        }
+    }
+
+    public void onAuthenticationComplete(Task<AuthResult> task) {
+        _registerProgressBar.setVisibility(View.GONE);
+
+        if (task.isSuccessful()) {
+            Toast.makeText(Register.this, "Account created", Toast.LENGTH_SHORT).show();
+        } else {
+            // Registration failed, handle the error
+            try {
+                throw task.getException();
+            } catch (FirebaseAuthWeakPasswordException e) {
+                // Handle weak password error
+                String errorMessage = "Weak password";
+                Toast.makeText(getApplicationContext(), errorMessage, Toast.LENGTH_LONG).show();
+            } catch (FirebaseAuthInvalidCredentialsException e) {
+                // Handle invalid email error
+                String errorMessage = "Invalid email address";
+                Toast.makeText(getApplicationContext(), errorMessage, Toast.LENGTH_LONG).show();
+            } catch (FirebaseAuthUserCollisionException e) {
+                // Handle email already in use error
+                String errorMessage = "Email already in use";
+                Toast.makeText(getApplicationContext(), errorMessage, Toast.LENGTH_LONG).show();
+            } catch (Exception e) {
+                // Handle other exceptions
+                String errorMessage = "Registration failed";
+                Toast.makeText(getApplicationContext(), errorMessage, Toast.LENGTH_LONG).show();
+            }
+        }
+    }
+
+    private void setRegisterButtonListener() {
+        // Setting listener variable for the register button
+        _registerButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                String email, password;
+
+                _registerProgressBar.setVisibility(View.VISIBLE);
+                email = String.valueOf(_editTextEmail.getText());
+                password = String.valueOf(_editTextPassword.getText());
+
+                if(TextUtils.isEmpty(email)){
```

Comment:
You should use || (OR) here and make the message something like: "Email or Password is empty"

Reply:
Its combined now